### PR TITLE
feat(unlock-app) - Create lock new page

### PR DESCRIPTION
**Pull request by kalidiagne**
This PR adds lock creation new page. 

**Form**
Screenshot 2022-09-08 at 18 35 46

**Summary 1**
Screenshot 2022-09-08 at 18 37 02

**Summary 2**
Screenshot 2022-09-08 at 18 37 31

**Change Network**
Screenshot 2022-09-08 at 18 40 00

**Network Selection**
Screenshot 2022-09-08 at 18 39 35

**Balance warning 1**
Screenshot 2022-09-08 at 22 42 53

**Balance warning 2**
Screenshot 2022-09-08 at 22 43 06

**Confirm lock creation**
Screenshot 2022-09-09 at 12 24 40

**Create lock bar and new link**
Screenshot 2022-09-09 at 19 32 11

**Form validations**
Screenshot 2022-09-09 at 19 52 10

**Tx error**
Screenshot 2022-09-09 at 23 49 40

**Key price**
Screenshot 2022-09-10 at 00 43 51

Screenshot 2022-09-10 at 00 50 59

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
This is looking pretty good!

File: unlock-app/src/components/interface/lock/Create/useCreateLock.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      unlimitedDuration: true,
+      unlimitedQuantity: true,
+    },
+  })
+
+  return {
+    form,
+    values: form.getValues(),
+  }
+}
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/interface/lock/Create/useCreateLock.tsx]: Agree with you! not necessary!

File: unlock-app/src/components/interface/lock/Create/CreateLock.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        <div className="md:max-w-lg">
+          <CreateLockFormSummary lock={lock} network={network} />
+          <div className="flex flex-col justify-between w-full gap-4 px-12 mt-12">
+            <Button>Look good for me</Button>
+            <span
+              className="font-bold text-center cursor-pointer"
+              onClick={() => setStep('data')}
+            >
+              Back to Edit
+            </span>
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/interface/lock/Create/CreateLock.tsx]: Yes! ill add the new variant on UI

**Comment by kalidiagne** [comment]
**Creation status update**
https://user-images.githubusercontent.com/20865711/189408627-d334f4ed-beb6-404c-b053-f0a5142ad65d.mov

**Creation flow**
https://user-images.githubusercontent.com/20865711/189408926-55d539c1-cf62-4cfc-a8ef-867d3dc27fcb.mov

File: unlock-app/src/components/interface/lock/Create/CreateLock.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/interface/lock/Create/CreateLock.tsx]: Added!

**Comment by julien51** [comment]
Don't show "Block 0/12", that's not real and would only be confusing. I asked @ChialiTsai but to provide us an animation similar to the one on checkout... that will be better.

Also, can you show us what happens on the select for duration and what happens when you toggle chain? or select token?

File: unlock-app/src/components/interface/lock/Create/elements/KeyPrice.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+
+interface KeyPriceProps {
+  price?: number
+  currency: string
+}
+
+export const TokenImage = ({ currency }: { currency: string }) => (
+  <img src={`/images/svg/tokens/${currency.toUpperCase()}.svg`} alt="" />
```
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/interface/lock/Create/elements/KeyPrice.tsx]: Thk! for the suggestion, just replaced with the component

**Comment by kalidiagne** [comment]
@julien51 @searchableguy 

Is all in place, waiting feedback for the custom contract address. 
I will remove the to the new `create lock` page, so we can merge it and gives to all the team the ability to test the page and the new flow, and also to make sure the is no blocking issue before replace the current one.

Please check any screens, ill provide more if needed

**Comment by kalidiagne** [comment]
https://user-images.githubusercontent.com/20865711/189728274-dffcbb82-99a7-402b-bd46-e750ca273c31.mov

**Comment by julien51** [comment]
Can you add `days` after the `Duration` amount in the "confirmation" screen please.

File: unlock-app/src/components/interface/lock/Create/CreateLock.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  }
+}
+
+const CreateLock = ({ onSubmit, defaultValues }: CreateLockProps) => {
+  return (
+    <div>
+      <div className="grid gap-4 md:grid-cols-2 md:gap-28">
+        <div className="flex flex-col mx-auto md:max-w-lg">
+          <h4 className="mb-4 text-5xl font-bold">
+            Connect with your 1000 true fans
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/CreateLock.tsx:144]:
```suggestion
            Deploy your membership contract
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+                      >
+                        {({ checked }) => (
+                          <div className="flex items-center w-full gap-4">
+                            <Radio checked={checked} />
+                            <div className="relative flex items-center w-full gap-4">
+                              <label
+                                className="text-lg font-bold cursor-pointer"
+                                htmlFor=""
+                              >
+                                Key duration
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx:205]:
```suggestion
                                Key duration (days)
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+              {errors?.expirationDuration && (
+                <span className="absolute mt-2 text-xs text-red-700">
+                  Please enter amount of days.
+                </span>
+              )}
+            </div>
+
+            <div>
+              <label className="block px-1 mb-4 text-base" htmlFor="">
+                Quantity
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx:236]:
```suggestion
                Number of memberships you want to sell
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+                    {...register('keyPrice', {
+                      required: true,
+                      min: 0,
+                    })}
+                  />
+                </div>
+              </div>
+              {errors?.keyPrice && (
+                <span className="absolute mt-2 text-xs text-red-700">
+                  Require to have postive number here.
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx:330]:
```suggestion
                  Please select a positive number
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+                autoComplete="off"
+                placeholder="Lock Name"
+                {...register('name', {
+                  required: true,
+                  minLength: 3,
+                })}
+              />
+              {errors?.name && (
+                <span className="absolute text-xs text-red-700">
+                  A cool name is required.
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx:156]:
```suggestion
                  A name is required.
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx [hunk 9]
```diff
@@ -0,0 +1,343 @@
...
+                          </div>
+                        )}
+                      </RadioGroup.Option>
+                    </RadioGroup>
+                  )
+                }}
+              />
+              {errors?.maxNumberOfKeys && (
+                <span className="absolute mt-2 text-xs text-red-700">
+                  How many do you say?
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockForm.tsx:294]:
```suggestion
                  Please chose a number of memberships for your lock.
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockFormSummary.tsx [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+  network: number
+  showStatus?: boolean
+  transactionHash?: string
+}
+
+type DeployStatus = 'progress' | 'deployed' | 'txError'
+const DEPLOY_STATUS_MAPPING: Record<DeployStatus, DeployStatusProps> = {
+  progress: {
+    title: 'This will take few minutes...',
+    description: 'Feel free to wait in this screen or return to main page.',
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockFormSummary.tsx:29]:
```suggestion
    description: 'Feel free to wait here or return to main page.',
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockFormSummary.tsx [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+    status: 'Progressing...',
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockFormSummary.tsx:30]:
```suggestion
    status: 'In progress...',
```

File: unlock-app/src/components/interface/lock/Create/elements/CreateLockFormSummary.tsx [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+                <span className="text-lg font-bold">{status}</span>
+              </div>
+              {transactionDetailUrl && (
+                <a
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className="inline-flex items-center gap-3 mt-5 ml-auto text-lg font-bold text-brand-ui-primary"
+                  href={transactionDetailUrl}
+                >
+                  <span>See on Etherscan</span>
```
Comment (julien51) [inline unlock-app/src/components/interface/lock/Create/elements/CreateLockFormSummary.tsx:148]:
```suggestion
                  <span>See on block explorer</span>
```

**Review by julien51 — APPROVED** [review]
It's a big PR but I think that's overall good. Once merged, please let's do extensive testing on staging!

**Comment by kalidiagne** [comment]
> Can you add `days` after the `Duration` amount in the "confirmation" screen please.

Screenshot 2022-09-13 at 10 56 01

Also updated the labels for form

Screenshot 2022-09-13 at 10 53 11

File: unlock-app/src/components/interface/lock/Create/CreateLock.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kalidiagne) [reply unlock-app/src/components/interface/lock/Create/CreateLock.tsx]: not missed, already present

**Pull request merged**